File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/likesdata/models/TumblrMetaVO.java
package nl.acidcats.tumblrlikes.data_impl.likesdata.models;

import com.google.auto.value.AutoValue;
import com.squareup.moshi.Json;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

/**
 * Created by stephan on 28/03/2017.
 */

@AutoValue
public abstract class TumblrMetaVO {
    public abstract int status();

    @Json(name = "msg")
    public abstract String message();

    public static JsonAdapter<TumblrMetaVO> jsonAdapter(Moshi moshi) {
        return new AutoValue_TumblrMetaVO.MoshiJsonAdapter(moshi);
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/likesdata/models/TumblrMetaVO.kt
package nl.acidcats.tumblrlikes.data_impl.likesdata.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Created on 24/10/2018.
 */
@JsonClass(generateAdapter = true)
data class TumblrMetaVO(
        val status: Int,
        @Json(name = "msg")
        val message: String
)
